src/Sql/Sql/ManagedDatabase/Cmdlet/MoveCopy/GetMoveCopyAzureSqlManagedDatabaseOperationsBase.cs

```diff
@@ -0,0 +1,105 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.ManagedDatabase.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Management.Automation;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet
+{
+    public abstract class GetMoveCopyAzureSqlManagedDatabaseOperationsBase : AzureSqlManagedDatabaseCmdletBase<IList<ManagedDatabaseMoveCopyOperation>>
+    {
+        protected const string GetMoveCopyManagedDatabaseOperationsByNameParameterSet = "GetMoveCopyManagedDatabaseOperationsByNameParameterSet";
+        protected const string GetMoveCopyManagedDatabaseOperationsByInputObjectParameterSet = "GetMoveCopyManagedDatabaseOperationsByInputObjectParameterSet";
+        protected const string GetMoveCopyManagedDatabaseOperationsByResourceIdParameterSet = "GetMoveCopyManagedDatabaseOperationsByResourceIdParameterSet";
+        protected const string GetMoveCopyManagedDatabaseOperationsByMoveCopyObjectParameterSet = "GetMoveCopyManagedDatabaseOperationsByMoveCopyObjectParameterSet";
+
+        [Parameter(Mandatory = false, HelpMessage = "The name of the instance database.", ParameterSetName = GetMoveCopyManagedDatabaseOperationsByNameParameterSet)]
```

Comment:
are we okay with no arg being mandatory for example for `GetMoveCopyAzureSqlManagedDatabaseOperationsBase`

Reply:
InstanceName and ResourceGroupName are required, and they are inherited from base classes.